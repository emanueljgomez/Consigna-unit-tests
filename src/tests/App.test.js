import React from "react";
import { shallow } from "enzyme";
import App from "../App.js";

describe("Test suite Cartas obtener usuarios", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  // Test Case 'a'
  test("debe coincidir con snapshot", () => {
    //const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
