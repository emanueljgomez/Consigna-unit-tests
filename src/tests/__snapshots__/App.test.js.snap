// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite Cartas obtener usuarios debe coincidir con snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Listado de Usuarios
        </h2>,
        <div
          className="row"
        >
          <p>
            Cargando página...
          </p>
        </div>,
      ],
      "className": "container",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "#ccffff",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Listado de Usuarios",
        },
        "ref": null,
        "rendered": "Listado de Usuarios",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Cargando página...
          </p>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cargando página...",
          },
          "ref": null,
          "rendered": "Cargando página...",
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Listado de Usuarios
          </h2>,
          <div
            className="row"
          >
            <p>
              Cargando página...
            </p>
          </div>,
        ],
        "className": "container",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "#ccffff",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Listado de Usuarios",
          },
          "ref": null,
          "rendered": "Listado de Usuarios",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Cargando página...
            </p>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cargando página...",
            },
            "ref": null,
            "rendered": "Cargando página...",
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
